consumer/pdata/generated_log.go

```diff
@@ -143,6 +143,26 @@ func (es ResourceLogsSlice) MoveAndAppendTo(dest ResourceLogsSlice) {
 	*es.orig = nil
 }
 
+// RemoveIf calls f sequentially for each element present in the slice.
+// If f returns true, the element is removed from the slice.
+func (es ResourceLogsSlice) RemoveIf(f func(ResourceLogs) bool) {
+	newLen := 0
```

Comment:
In Go, you don't need to implement this logic but can use the built in copy.

s := (*es.orig)
copy(s[i:], s[i+1:])
s = s[:len(s)-1]